### PR TITLE
Make the nwcsaf nc readers tests use real files

### PR DESCRIPTION
**Pull request by mraspaud**
This PR creates actual files for testing the NWCSAF NC reader. This will allow using these files in other contexts. The tests were refactored along the way, and tests for the start and end time properties and attributes were added.

 - [x] Tests added

### CODE REVIEW

File: satpy/tests/reader_tests/test_nwcsaf_nc.py [hunk 1]
```diff
@@ -16,84 +16,271 @@
...
+def nwcsaf_geo_ct_filehandler(nwcsaf_geo_ct_filename):
+    """Create a CT filehandler."""
+    return NcNWCSAF(nwcsaf_geo_ct_filename, {}, {})
+
+
+@pytest.fixture(scope="session")
+def nwcsaf_pps_cmic_filename(tmp_path_factory):
+    """Create a CMIC file."""
+    filename = tmp_path_factory.mktemp("data") / "S_NWC_CMIC_npp_00000_20230118T1427508Z_20230118T1429150Z.nc"
+    with h5netcdf.File(filename, mode="w") as nc_file:
```
Comment (adybbroe) [inline satpy/tests/reader_tests/test_nwcsaf_nc.py:114]: Maybe you could create a small function to create the actual file, which could be used when creating the cpp file as well, so we reduce duplication?
```suggestion
    create_pps_cloud_microphysics_file(filename, 'cmic')
```

File: satpy/readers/nwcsaf_nc.py [hunk 2]
```diff
@@ -414,3 +407,16 @@ def remove_empties(variable):
...
     return variable
+
+
+def read_nwcsaf_time(time_value):
+    """Read the time, nwcsaf-style."""
+    try:
+        # MSG:
+        try:
+            return datetime.strptime(time_value, '%Y-%m-%dT%H:%M:%SZ')
+        except TypeError:
```
Comment (adybbroe) [inline satpy/readers/nwcsaf_nc.py:418]: Is that possibly for older NWCSAF files? Would be good to have it tested in that case.

**Review by adybbroe — COMMENTED** [review]
Many thanks for this. Thie is really nice and can be used also for testing other applications loading a scene with Satpy.

File: satpy/readers/nwcsaf_nc.py (hunk 2 above)
Comment (mraspaud) [reply satpy/readers/nwcsaf_nc.py:418]: Ok, so looking at this, I think both recent version of netcdf and h5netcdf do the conversion to string automatically, so this part can be removed in the future I think. For now, I can't reproduce a netcdf that requires this, so I will leave this untested unfortunately.

File: satpy/tests/reader_tests/test_nwcsaf_nc.py (hunk 1 above)
Comment (mraspaud) [reply satpy/tests/reader_tests/test_nwcsaf_nc.py:128]: Done!

**Review by adybbroe — APPROVED** [review]
LGTM.

**Pull request merged**